source/event_stream_rpc_client.c

```diff
@@ -326,14 +359,18 @@ void aws_event_stream_rpc_client_connection_close(
         (void *)connection,
         aws_error_debug_str(shutdown_error_code));
 
-    if (aws_event_stream_rpc_client_connection_is_open(connection)) {
-        aws_atomic_store_int(&connection->is_open, 0U);
+    size_t expect_open = 1U;
+    if (aws_atomic_compare_exchange_int(&connection->is_open, &expect_open, 0U)) {
         aws_channel_shutdown(connection->channel, shutdown_error_code);
 
         if (!connection->bootstrap_owned) {
             aws_mutex_lock(&connection->stream_lock);
-            aws_hash_table_clear(&connection->continuation_table);
+            aws_hash_table_foreach(&connection->continuation_table, s_mark_each_continuation_closed, NULL);
```

Comment:
This is a carbon copy of the above code, isn't it? Should we make a function out of this?

Reply:
Yeah, an earlier iteration I had was doing weirder stuff with locks, but this can be functionized